lang/cs/Org.Apache.REEF.Client/AzureBatch/Storage/AzureStorageClient.cs

```diff
@@ -0,0 +1,109 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Org.Apache.REEF.Client.AzureBatch.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Client.AzureBatch.Storage
+{
+    internal sealed class AzureStorageClient
+    {
+        private static readonly Logger LOGGER = Logger.GetLogger(typeof(AzureStorageClient));
+        private static readonly string StorageConnectionStringFormat = "DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}";
+        private static readonly int SASTokenValidityMinutes = 60;
+
+        private readonly string _storageAccountName;
+        private readonly string _storageAccountKey;
+        private readonly string _storageContainerName;
+
+        private readonly string _storageConnectionString;
+
+        [Inject]
+        AzureStorageClient(
+            [Parameter(typeof(AzureStorageAccountName))] string storageAccountName,
+            [Parameter(typeof(AzureStorageAccountKey))] string storageAccountKey,
+            [Parameter(typeof(AzureStorageContainerName))] string storageContainerName)
+        {
+            this._storageAccountName = storageAccountName;
+            this._storageAccountKey = storageAccountKey;
+            this._storageContainerName = storageContainerName;
+
+            this._storageConnectionString = string.Format(StorageConnectionStringFormat,
+                new object[] { storageAccountName, storageAccountKey });
+        }
+
+        /// <summary>
+        /// Uploads a given file to the given destination folder in Azure Storage.
+        /// </summary>
+        /// <param name="destination">Destination in Azure Storage where given file will be uploaded.</param>
+        /// <param name="filePath">Path to the file to be uploaded.</param>
+        /// <returns>Storage SAS URI for uploaded file.</returns>
+        public async Task<Uri> UploadFile(string destination, string filePath)
+        {
+            CloudBlobContainer blobContainer = await this.GetOrCreateCloudBlobContainer();
+            CloudBlobDirectory directory = blobContainer.GetDirectoryReference(destination);
+
+            string fileName = Path.GetFileName(filePath);
+            CloudBlockBlob blob = directory.GetBlockBlobReference(fileName);
+            await blob.UploadFromFileAsync(filePath);
+
+            string sas = blob.GetSharedAccessSignature(CreateSASPolicy());
+            string uri = blob.Uri.AbsoluteUri;
+            Uri uploadedFile = new Uri(uri + sas);
+            LOGGER.Log(Level.Info, "Uploaded {0} jar file to {1}", new string[] { filePath, uploadedFile.ToString() });
```

Comment:
no need to call `.ToString()` explicitly in logging

Reply:
"Cannot implicitly convert type System.Uri to string". I think it should be called as it is not being put into "Logger.Log" but "new string[]"

Reply:
oh, of course it should be simply
```C#
LOGGER.Log(Level.Info, "Uploaded {0} jar file to {1}", filePath, uploadedFile);
```
- we don't need to pack arguments into an array like we do in Java.

Reply:
you are right:) updated.